docs/DeveloperGuide.adoc

```diff
@@ -570,22 +565,284 @@ image::autocomplete.png[width="500"]
 
 // end::autocompleteDG[]
 
-// tag::addappointment[]
-=== Adding & Storing Appointments
+// tag::find[]
+=== Finding a Contact or Pet Patient
+The find function allows users to find for a specific contact or pet patient through the listed fields.
+
+==== Current Implementation
+
+==== Design Consideration
+
+===== Aspect:
+* **Alternative 1:**
+** Pros:
+** Cons:
+
+* **Alternative 2 (current choice):**
+** Pros:
+** Cons:
+
+==== Current Limitations
+Users are unable to find appointment by type as the CalendarFX runs differently from the other classes.
+
+==== Future Considerations
+//end::find[]
+
+// tag::listappt[]
+=== Listing Appointments of Different Views and Dates
+The command allows users to list appointments based on the date that they wish to see, and the results are displayed via the API of CalendarFX.
+
+==== Current Implementation
+After Medeina implemented CalendarFX, users are able to view appointments at a specified date by clicking on the Calendar. However, as Medeina is a CLI-based application, users should not be required to move their mouse. Thus, there was a need to be able to view appointments of specific dates through the command line.
+
+.Day view of the Calendar obtained by the `listappt -d` command.
+image::ListAppointmentDayView.png[width="800"]
+
+The implemented CalendarFX API consists of 4 different views: Day, Week, Month and Year views. Users should be able to get a specific requested view of a date, month, or year they wish to see.
+
+The below diagram illustrates the interaction between the `UI`, `Logic` and `Events` components in the `ListAppointment` command.
+
+.Sequence diagram for interactions between the different components upon running the `listappt -y 2017` command.
+image::SDforListAppointment.png[width="800"]
+
+`ListAppointmentParser.java` handles and separates the request of users based on a regular expression that captures the matching groups of options `-d`, `-w`, `-m`, `-y`. The parser then checks the remaining user input, and parses it into `LocalDate`, `YearMonth` and `Year` classes. Should there be no remaining user input after the option, the current date is obtained as the field. A new `ListAppointmentCommand` is created with the type (1 for year, 2 for month, 3 for week, 4 for date) as well as the parsed class.
+
+In `execute()`, there is a switch that determines the type of command run based on the type obtained from the parser. The relevant functions to change the Calendar gets called based on the type, and an event gets called to handle the change. A check is done to determine whether there were appointments in the year of the requested date, so that users are not able to randomly jump into years that should not have ever had any appointments (i.e. years before Medeina existed).
```

Comment:
whether there "are".
Would it be better to just say "jump into years that should not have appointments" ?